tools/Datadog.Core.Tools/TcpPortProvider.cs

```diff
@@ -17,6 +17,7 @@ public static class TcpPortProvider
     {
         private static readonly object PortLock = new { };
         private static readonly ConcurrentBag<int> ReturnedPorts = new ConcurrentBag<int>();
+        [ThreadStatic]
```

Comment:
`MinPort` will only be initialized by the first thread to touch this class (in the static ctor). Other threads will see the default value (`0`). For `ThreadStatic` to work that way you intend, each thread has to initialize the field's value.

Reply:
Thanks for the heads up! I was trying to improve the distribution of starting points by making it ThreadStatic, but Colin suggested a new way to grab free ports so I'm trying that right now